File: client/lib/common/models/meeting_review.dart
class MeetingReview {
  bool anonymous = true;

  int objective = -1;
  int smoothRunning = -1;
  int preparation = -1;
  int length = -1;
  int respect = -1;

  String comments = "";

  MeetingReview();

  factory MeetingReview.fromJson(Map<String, dynamic> json) {
    return MeetingReview()
      ..anonymous = json['isAnonymous'] as bool? ?? true
      ..objective = json['objective'] as int? ?? -1
      ..smoothRunning = json['smoothRunning'] as int? ?? -1
      ..preparation = json['preparation'] as int? ?? -1
      ..length = json['length'] as int? ?? -1
      ..respect = json['respect'] as int? ?? -1
      ..comments = json['comments'] as String? ?? "";
  }

  Map<String, dynamic> toJson() {
    return {
      'isAnonymous': anonymous,
      'objective': objective,
      'smoothRunning': smoothRunning,
      'preparation': preparation,
      'length': length,
      'respect': respect,
      'comments': comments,
    };
  }

  static bool isValidGrade(int value) => value >= 1 && value <= 5;

  bool get hasValidRatings =>
      [objective, smoothRunning, preparation, length, respect]
          .every(isValidGrade);
}

File: server/flaskr/models/data/meeting_review.py
from dataclasses import dataclass

@dataclass(slots=True)
class MeetingReview:
    meetingId: int
    userId: int

    isAnonymous: bool
    objective: int
    smoothRunning: int
    preparation: int
    length: int
    respect: int

    comments: str

    def to_dict(self):
        return {
            'meetingId': self.meetingId,
            'userId': self.userId,

            'isAnonymous': self.isAnonymous,
            'objective': self.objective,
            'smoothRunning': self.smoothRunning,
            'preparation': self.preparation,
            'length': self.length,
            'respect': self.respect,

            'comments': self.comments,
        }
